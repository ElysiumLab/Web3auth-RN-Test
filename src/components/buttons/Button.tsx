import {
  TouchableOpacity,
  Text,
  Image,
  StyleSheet,
  TextStyle,
} from "react-native";

import { COLORS, SIZES } from "../../styles";

export const CircleButton = ({
  imgUrl,
  handlePress,
  styleTouchable,
  styleImage,
}: any) => {
  return (
    <TouchableOpacity
      style={[styles.circleButtonTouchable, styleTouchable]}
      accessibilityRole="imagebutton"
      onPress={handlePress}
    >
      <Image
        style={[styles.circleButtonImage, styleImage]}
        source={imgUrl}
        resizeMode="contain"
      />
    </TouchableOpacity>
  );
};

export const RectButton = ({
  title,
  styleTouchable,
  styleText,
  handlePress,
}: any) => {
  return (
    <TouchableOpacity
      style={[styles.rectButtonTouchable, styleTouchable]}
      onPress={handlePress}
    >
      <Text style={[styles.rectButtonText, styleText]}>{title}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  circleButtonTouchable: {
    width: 40,
    height: 40,
    // backgroundColor: COLORS.black,
    position: "absolute",
    borderRadius: SIZES.extraLarge,
    justifyContent: "center",
    alignItems: "center",
  },

  circleButtonImage: {
    tintColor: COLORS.white,
  },

  rectButtonTouchable: {
    justifyContent: "center",
  },

  rectButtonText: {
    textAlign: "center",
  },
});
