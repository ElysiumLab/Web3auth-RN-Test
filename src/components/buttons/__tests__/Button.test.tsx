import React from "react";
import { render, fireEvent, screen } from "@testing-library/react-native";

import { RectButton, CircleButton } from "../Button";

describe("<RectButton />", () => {
  it("renders correctly", () => {
    const tree = render(<RectButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("press correctly", async () => {
    const mockCallback = jest.fn();
    const title = "testRectButton";

    render(<RectButton title={title} handlePress={mockCallback} />);

    const rectButton = await screen.findByText(title);

    fireEvent.press(rectButton);

    expect(mockCallback).toHaveBeenCalled();
  });
});

describe("<CircleButton />", () => {
  it("renders correctly", () => {
    const tree = render(<CircleButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("press correctly", async () => {
    const mockCallback = jest.fn();

    render(<CircleButton handlePress={mockCallback} />);

    const circleButton = await screen.findByRole("imagebutton");

    fireEvent.press(circleButton);

    expect(mockCallback).toHaveBeenCalled();
  });
});
