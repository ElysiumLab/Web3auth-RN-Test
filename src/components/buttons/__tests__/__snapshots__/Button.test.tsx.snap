// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CircleButton /> renders correctly 1`] = `
<View
  accessibilityRole="imagebutton"
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 24,
      "height": 40,
      "justifyContent": "center",
      "opacity": 1,
      "position": "absolute",
      "width": 40,
    }
  }
>
  <Image
    resizeMode="contain"
    style={
      Array [
        Object {
          "tintColor": "#FFFFFF",
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`<RectButton /> renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "textAlign": "center",
        },
        undefined,
      ]
    }
  />
</View>
`;
